spec/bundler_inject_spec.rb

```diff
@@ -0,0 +1,297 @@
+        end
+
+        it "with global file" do
+          write_global_bundler_d_file <<~F
+            gem "rack-obama"
+          F
+          bundle(:update)
+
+          expect(lockfile_specs).to match_array [["rack", "2.0.6"], ["rack-obama", "0.1.1"]]
+        end
+
+        it "with local and global files" do
+          write_bundler_d_file <<~F
+            gem "rack-obama"
+          F
+          write_global_bundler_d_file <<~F
+            gem "omg"
+          F
+          bundle(:update)
+
+          expect(lockfile_specs).to match_array [["rack", "2.0.6"], ["rack-obama", "0.1.1"], ["omg", "0.0.6"]]
+        end
+      end
+
+      describe "#override_gem" do
+        it "with a different version" do
+          write_bundler_d_file <<~F
+            override_gem "rack", "=2.0.5"
+          F
+          bundle(:update)
+
+          expect(lockfile_specs).to eq [["rack", "2.0.5"]]
+          expect(err).to match %r{^\*\* override_gem\("rack", "=2.0.5"\) at .+/bundler\.d/local_overrides.rb:1$}
+        end
+
+        it "with a git repo" do
+          write_bundler_d_file <<~F
+            override_gem "rack", :git => "https://github.com/rack/rack"
+          F
+          bundle(:update)
+
+          expect(lockfile_specs).to eq [["rack", extract_rack_version]]
+          expect(err).to match %r{^\*\* override_gem\("rack", :git=>"https://github.com/rack/rack"\) at .+/bundler\.d/local_overrides.rb:1$}
+        end
+
+        it "with a path" do
+          with_path_based_gem("https://github.com/rack/rack") do |path|
+            write_bundler_d_file <<~F
+              override_gem "rack", :path => #{path.to_s.inspect}
+            F
+            bundle(:update)
+
+            expect(lockfile_specs).to eq [["rack", extract_rack_version(path)]]
+            expect(err).to match %r{^\*\* override_gem\("rack", :path=>#{path.to_s.inspect}\) at .+/bundler\.d/local_overrides.rb:1$}
+          end
+        end
+
+        it "with a path that includes ~" do
+          with_path_based_gem("https://github.com/rack/rack") do |path|
+            path = Pathname.new("~/#{path.relative_path_from(Pathname.new("~").expand_path)}")
+
+            write_bundler_d_file <<~F
+              override_gem "rack", :path => #{path.to_s.inspect}
+            F
+            bundle(:update)
+
+            expect(lockfile_specs).to eq [["rack", extract_rack_version(path)]]
+            expect(err).to match %r{^\*\* override_gem\("rack", :path=>#{path.expand_path.to_s.inspect}\) at .+/bundler\.d/local_overrides.rb:1$}
+          end
+        end
+
+        it "when the gem doesn't exist" do
+          write_bundler_d_file <<~F
+            override_gem "omg"
+          F
+          bundle(:update, expect_error: true)
+
+          stream = bundler_short_version == "2.0" ? err : out
+          expect(stream).to include "Trying to override unknown gem \"omg\""
+        end
+      end
+
+      describe "#ensure_gem" do
```

Comment:
After we added the global bundler.d, I ran into this problem almost immediately because I added "pry" to my Gemfile, and one of the other projects I am testing this in *already* had pry in the Gemfile.